synchdb.c

```diff
@@ -2046,6 +2046,35 @@ dbz_mark_batch_complete(int batchid)
 	return 0;
 }
 
+static void
+remove_dbz_metadata_files(const char * name)
+{
+	struct dirent *entry;
+	char filepath[256] = {0};
+	char keyword[SYNCHDB_CONNINFO_NAME_SIZE] = {0};
+
+	DIR *dir = opendir(SYNCHDB_METADATA_DIR);
+	if (!dir)
+		elog(ERROR, "failed to open synchdb metadata dir %s", SYNCHDB_METADATA_DIR);
+
+	snprintf(keyword, SYNCHDB_CONNINFO_NAME_SIZE, "_%s_", name);
+
+	while ((entry = readdir(dir)) != NULL)
+	{
+		if (strcmp(entry->d_name, ".") == 0 || strcmp(entry->d_name, "..") == 0)
+			continue;
+
+		if (strstr(entry->d_name, keyword) != NULL)
+		{
+			elog(WARNING, "Deleting metadata file: %s", filepath);
```

Comment:
the filepath hasn't been initialized with the current file yet. This will log incorrect information.